### ISSUE TITLE
Trouble opening fish_config

### ISSUE THREAD

**Issue by willhansen** [fish-shell/fish-shell#9014]
fish version: 3.3.1

Linux my-pc 5.15.0-35-generic #36-Ubuntu SMP Sat May 21 02:24:07 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

Only thing I installed is oh-my-fish

No difference with `sh -c 'env HOME=$(mktemp -d) fish'`

using `fish_config` gives a "file not found" error with firefox version 100.0.2

if I `cat` the file, then open the href link in the file (with the same browser), everything works fine.

**Comment by faho** [comment]
Unfortunately,new Ubuntu versions ship firedox as a snap without access to /tmp. There is nothing we can do.

**Comment by faho** [comment]
Duplicate of #8844

**Comment by zanchey** [comment]
You can work around this, I think, by setting TMPDIR to something in your home directory.

**Comment by faho** [comment]
Okay, I've made it print the real URL. Unfortunately we can't use it directly because that's a literal CVE (it can't be used as a commandline argument or other users can find it), and we can't really use other paths (they're all wrong somewhere), and we can't even really *detect* if the browser has been snap-broken.

Really, the only realistic fix is for Ubuntu to fix either xdg-open (so it does something like flatpak portals) or allow /tmp.

**Comment by willhansen** [comment]
How about a command line option for fish_config to specify a tmp directory?

`fish_config --tmpdir ~/Downloads/firefox.tmp`

The python tempfile module supports it: https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile

**Comment by faho** [comment]
>How about a command line option for fish_config to specify a tmp directory?

That doesn't really give you any more capabilities. You can already set $TMPDIR - `TMPDIR=~/Downloads/firefox.tmp fish_config`.

You'd still have to do it before starting fish_config. We could point towards $TMPDIR, but the explanation would be annoying: "If your system doesn't allow your browser to access the path we asked it to open in a browser, you can set $TMPDIR to a path it is allowed to open". With "--tmpdir" that just changes to "give the --tmpdir option with a path it is allowed to open", which isn't really an improvement.

So you can simply click the url we now print - after you figured that /tmp isn't working, you click that and that works.
